packages/classes/src/PayloadArray.js

```diff
@@ -0,0 +1,85 @@
+import { sortedIndexBy } from './tools/utils/sortedArray'
+import Payload from './Payload'
+
+/**
+ * PayloadArray stores a series of Payloads in a data structure that allows for easy filtering
+ * and lookup by position and type.
+ */
+class PayloadArray {
+	payloads = []
+
+	constructor(...payloads) {
+		payloads.forEach((payload) => this.insert(payload))
+	}
+
+	get length() { return this.payloads.length }
+
+	/**
+	 * Insert a Payload into the PayloadArray while maintaining position order.
+	 *
+	 * @param  {Payload} payload The Payload to be inserted.
+	 * @return {PayloadArray}    This PayloadArray to allow for chaining.
+	 */
+	insert = (payload) => {
+		if (!(payload instanceof Payload)) {
+			throw new Error('Attempt to insert a non-Payload to PayloadArray')
+		}
+		const index = this.indexOfPosition(payload.position)
+		this.payloads.splice(index, 0, payload)
+		return this
+	}
+
+	/**
+	 * Find the index to which a payload with a given position would be inserted.
+	 *
+	 * @param  {Number}  position      The position being searched for.
+	 * @param  {Boolean} returnHighest The position being searched for.
```

Comment:
Description for this param is copypasta.

Reply:
How can you be sure?

Reply:
Just a natural genius I guess.